### ISSUE TITLE
Home page header flashes when scrolled to a certain position

### ISSUE THREAD

**Issue by lneugebauer** [benjaminkott/bootstrap_package#1424]
# Bug Report

## Prerequisites

* [ ] Can you reproduce the problem on TYPO3 v11.5 LTS
* [ ] Can you reproduce the problem on TYPO3 v12.4 LTS
* [x] Did you [perform a cursory search](https://github.com/issues?q=is%3Aissue+user%3Abenjaminkott+repo%3Abenjaminkott%2Fbootstrap_package)
  to see if your bug or enhancement is already reported?

## Description

When slowly scrolling on [bootstrap-package.com](https://www.bootstrap-package.com/) the page header starts flashing. See attached screen recording.

## Steps to Reproduce

1. Open [bootstrap-package.com](https://www.bootstrap-package.com/)
2. Slowly scroll down
3. See strobe effect

### Expected behavior

The animation should probably be started at the starting pixel threshold and not be constrained to the current scroll position.

### Actual behavior

The animation is constrained to the scroll position which causes the page header to flicker.

### Screenshots

https://github.com/benjaminkott/bootstrap_package/assets/25110308/99b3b319-437d-42d6-9ce9-b64a048ccd0d

## Versions

https://www.bootstrap-package.com/

**Comment by Hawkeye1909** [comment]
I've had this flickering in a project as well. The problem is the mechanism by which the class 'navbar-transition' is added. I ended up editing the funtion animateHeader in bootstrap.stickyheader.js to look like this:
```
function animateHeader() {
    if (window.scrollY >= stickyheader[0].getBoundingClientRect().bottom && !stickyheader[0].classList.contains("navbar-transition")) {
        stickyheader[0].classList.add("navbar-transition");
    } else if (window.scrollY < stickyheader[0].getBoundingClientRect().bottom && stickyheader[0].classList.contains("navbar-transition")) {
        stickyheader[0].classList.remove("navbar-transition");
    }
}
```

**Comment by bznovak** [comment]
@Hawkeye1909 This fix did not work for me.

**Comment by sukitsubaki** [comment]
I appreciate the suggestions, but I'm wondering if there might be a better way to address the root cause of the issue rather than just overriding certain behaviors with JavaScript. Simply replacing the existing script feels more like a workaround than a proper fix.

I already tried adjusting the CSS: `.navbar { transition: all 1s cubic-bezier(0, 0, 0.58, 1) 0s !important; }`

Unfortunately, this didn't solve the problem either. Interestingly, the issue only occurs in Chromium-based browsers like Chrome and Microsoft Edge, while everything works fine in Safari and Firefox. This suggests that the problem might be related to how Chromium handles scrolling and animations.

See also #1468 (duplicate issue)

**Comment by CybotTM** [comment]
CSS-only root cause fix posted in the duplicate issue #1468: https://github.com/benjaminkott/bootstrap_package/issues/1468#issuecomment-4061144764

TL;DR: The root cause is that `position: sticky` + navbar height change = document height change. Adding compensating `margin-bottom` on `.navbar-transition` keeps document height constant. Pure CSS, no JS changes needed.
